src/main/java/traincraft/blocks/assemblytables/ContainerAssemblyTable.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("null.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+package traincraft.blocks.assemblytables;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.InventoryBasic;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+import net.minecraftforge.items.SlotItemHandler;
+
+//TODO: BUG: Clicking too far above or below (where a normal minecraft GUI would not be) drops the items out of inventory.
+// Change collision box/safe area or whatever
+public class ContainerAssemblyTable extends Container {
+
+    private final World world;
+    private final TileAssemblyTable tileAssemblyTable;
+
+    private InventoryBasic outputInventory = new InventoryBasic("output inv", false, 8);
+    
+    public ContainerAssemblyTable(InventoryPlayer playerInventory, World thisWorld, TileAssemblyTable tileAssemblyTable) {
+        this.world = thisWorld;
+        this.tileAssemblyTable = tileAssemblyTable;
+
+        //This offset controls how up-and-down the slots are, I did this because I measured it all out on the image, but it doesn't fit to the Gui, so this.
+        int offset = -45;
```

Comment:
This isn't very nice and also your problem is that you didn't set the xSize and ySize in the gui constructor. This would fix your out of bounds and this offset bug

Reply:
i see, thank you, it is now fixed